src/json/get-meta-from-path.js

```diff
@@ -1,11 +1,28 @@
-import parse, { getMeta, getValue } from './parse';
-
-export default function getMetaFromPath(jsonString, dataPath) {
+export default function getMetaFromPath(
+  jsonAst,
+  dataPath,
+  isIdentifierLocation
+) {
   // TODO: Handle json pointer escape notation and better error handling
-  return getMeta(
-    dataPath
-      .split('/')
-      .slice(1)
-      .reduce((obj, pointer) => getValue(obj, pointer), parse(jsonString))
-  );
+  const pointers = dataPath.split('/').slice(1);
+  const lastPointerIndex = pointers.length - 1;
+  return pointers.reduce((obj, pointer, idx) => {
+    switch (obj.type) {
+      case 'Object': {
+        const filtered = obj.children.filter(
+          child => child.key.value === pointer
+        );
+        if (filtered.length !== 1) {
+          throw new Error(`Couldn't find property ${pointer} of ${dataPath}`);
+        }
+        const { key, value } = filtered[0];
+        return isIdentifierLocation && idx === lastPointerIndex ? key : value;
```

Comment:
Is this `isIdentifierLocation` determines whether the ☝️ is pointing at the `key` or the `value`?

Reply:
I see I see, just the variable name a bit confusing to me. Don't know what it means by just looking at it `isIdentifierLocation`.

Reply:
Yeah, you are right! I copied it from the caller `getLocation`. But, it doesn't make much in this _context_. Do you have any suggestion?